### PR TITLE
CXP-311: update error messages

### PR DESCRIPTION
**Pull request by pchasle**
- Category doesn’t exist
- Attribute doesn’t exist
- Family doesn’t exist
- Product doesn’t exist

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/ErrorManagement/CollectApiError/CollectProductDomainErrorEndToEnd.php [hunk 1]
```diff
@@ -279,4 +311,110 @@ public function test_it_collects_an_unknown_attribute_error(): void
...
+
+        $client->request('PATCH', '/api/rest/v1/products/high-top_sneakers', [], [], [], $content);
+        Assert::assertSame(Response::HTTP_UNPROCESSABLE_ENTITY, $client->getResponse()->getStatusCode());
+
+        $this->elasticsearch->refreshIndex();
+        $result = $this->elasticsearch->search([]);
+
+        Assert::assertCount(1, $result['hits']['hits']);
+
+        $doc = $result['hits']['hits'][0]['_source'];
```
Comment (pierrejolly) [inline src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/ErrorManagement/CollectApiError/CollectProductDomainErrorEndToEnd.php:344]: I don't understand the naming :/ ,why did you use `$doc` ?

Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/ErrorManagement/CollectApiError/CollectProductDomainErrorEndToEnd.php:344]: Elasticsearch store _documents_ = doc, and that come from the elasticsearch examples where they use `$doc` with their phpclient

File: src/Akeneo/Connectivity/Connection/back/Application/ErrorManagement/Service/CollectApiError.php [hunk 2]
```diff
@@ -57,7 +57,7 @@ public function __construct(
     }
 
     public function collectFromProductDomainError(
-        ProductInterface $product,
+        ?ProductInterface $product,
```
[the remark this answers is not in the document]
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/back/Application/ErrorManagement/Service/CollectApiError.php:60]: Indeed, but I didn't want to add too many modifications to this PR.
I can still add it now though.

[the remark this answers is not in the document]
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/back/Application/ErrorManagement/Service/CollectApiError.php]: Done, and I added missing tests

**Pull request merged**